lib/datadog/tracing/span_event.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require 'time'
+
+module Datadog
+  module Tracing
+    # SpanEvent represents an annotation on a span.
+    # @public_api
+    class SpanEvent
+      # @!attribute [r] name
+      #   @return [Integer]
+      attr_reader :name
+
+      # @!attribute [r] attributes
+      #   @return [Hash{String => String, Numeric, Boolean, Array<String, Numeric, Boolean>}]
+      attr_reader :attributes
+
+      # @!attribute [r] time_unix_nano
+      #   @return [Integer]
+      attr_reader :time_unix_nano
+
+      def initialize(
+        name,
+        attributes: nil,
+        time_unix_nano: nil
```

Comment:
Is there any usage, that caller does not provide a timestamp?

Reply:
In the OpenTelemetry API [Span.record_exception](https://github.com/open-telemetry/opentelemetry-ruby/blob/35abeac8653359aebd0231c94a23587ad4ee372e/sdk/lib/opentelemetry/sdk/trace/span.rb#L139) can be called without a timestamp. We can force the caller to always pass a timestamp and convert the time from seconds to nanoseconds but I think it's better to centralize that logic in `SpanEvent.new(...)`